zmon_agent/discovery/kubernetes/cluster.py

```diff
@@ -1129,3 +1142,76 @@ def get_postgresql_databases(cluster_id, alias, environment, region, infrastruct
                 entities.append(replicatype_entity)
 
     return entities
+
+
+@trace(tags={'kubernetes': 'hpa'}, pass_span=True)
+def get_cluster_hpas(kube_client, cluster_id, alias, environment, region, infrastructure_account,
+                     namespace=None, **kwargs) -> list:
+    current_span = extract_span_from_kwargs(**kwargs)
+    entities = []
+
+    hpas = get_all(kube_client, kube_client.get_hpas, namespace, span=current_span)
+
+    for h in hpas:
+        obj = h.obj
+
+        conditions_annotation = obj['metadata']['annotations']['autoscaling.alpha.kubernetes.io/conditions']
+        conditions_list = json.loads(conditions_annotation)
+        conditions = {c['type']: c['status'] == 'True' for c in conditions_list}
+
+        entity = {
+            'id': 'hpa-{}-{}[{}]'.format(h.name, h.namespace, cluster_id),
```

Comment:
is `h.name` unique?

Reply:
same for `cs.name` in crednetialsets.

Reply:
`h.name` and `h.namespace` is always unique.